src/ape_run/_cli.py

```diff
@@ -1,33 +1,155 @@
+from pathlib import Path
+from runpy import run_module
+from typing import Dict, Union
+
 import click
 
-from ape.cli import NetworkBoundCommand, ape_cli_context, network_option
+from ape.cli import NetworkBoundCommand, network_option
+from ape.logging import logger
+from ape.managers.project import ProjectManager
+from ape.utils import cached_property, get_relative_path, use_temp_sys_path
+from ape_console._cli import console
+
+
+class ScriptCommand(click.MultiCommand):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs, result_callback=self.result_callback)
+        self._namespace = {}
+
+    def _get_command(self, filepath: Path) -> Union[click.Command, click.Group, None]:
+        relative_filepath = get_relative_path(filepath, self._project.path)
+
+        # First load the code module by compiling it
+        # NOTE: This does not execute the module
+        logger.debug(f"Parsing module: {relative_filepath}")
+        try:
+            code = compile(filepath.read_text(), filepath, "exec")
+        except SyntaxError as e:
+            logger.error_from_exception(e, f"Exception while parsing script: {relative_filepath}")
+            return None  # Prevents stalling scripts
+
+        # NOTE: Introspect code structure only for given patterns (do not execute it to find hooks)
+        if "cli" in code.co_names:
+            # If the module contains a click cli subcommand, process it and return the subcommand
+            logger.debug(f"Found 'cli' command in script: {relative_filepath}")
+
+            with use_temp_sys_path(filepath.parent.parent):
+                try:
+                    ns = run_module(f"scripts.{filepath.stem}")
+                except Exception as e:
+                    logger.error_from_exception(
+                        e, f"Exception while parsing script: {relative_filepath}"
+                    )
+                    return None  # Prevents stalling scripts
+
+            self._namespace[filepath.stem] = ns
+            return ns["cli"]
+
+            return None  # NOTE: Allow other scripts to load if loading script breaks
```

Comment:
Does this actually ever get called?
Wouldn't the return statement on line 46 always either cause an exception to occur or actually return the CLI?

Reply:
Oh yeah, whoops. I found out that the `if` statement I had before was unreachable, forgot this line!